In [11]:
import pandas as pd

In [12]:
trade_data=pd.read_csv("comptab_2018-01-29 16_00_comma_separated.csv")

In [13]:
trade_data.head()

Year App.                 Taxon Class          Order        Family  \
0  2016    I        Aquila heliaca  Aves  Falconiformes  Accipitridae   
1  2016    I        Aquila heliaca  Aves  Falconiformes  Accipitridae   
2  2016    I  Haliaeetus albicilla  Aves  Falconiformes  Accipitridae   
3  2016    I  Haliaeetus albicilla  Aves  Falconiformes  Accipitridae   
4  2016    I  Haliaeetus albicilla  Aves  Falconiformes  Accipitridae   

        Genus Importer Exporter Origin  Importer reported quantity  \
0      Aquila       TR       NL     CZ                         NaN   
1      Aquila       XV       RS     RS                         NaN   
2  Haliaeetus       BE       NO    NaN                         NaN   
3  Haliaeetus       BE       NO    NaN                         NaN   
4  Haliaeetus       DK       IS    NaN                       700.0   

   Exporter reported quantity       Term Unit Purpose Source  
0                         1.0     bodies  NaN       T      C  
1                         1.0     bodies  NaN       Q      O  
2                        43.0   feathers  NaN       S      W  
3                        43.0  specimens  NaN       S      W  
4                         NaN  specimens  NaN       S      W

In [14]:
len(trade_data)

67161

In [15]:
trade_data_2016=trade_data[trade_data["Year"]==2016]

In [16]:
len(trade_data_2016)

67007

In [27]:
trade_data_2017=trade_data[trade_data["Year"]==2017]

In [28]:
len(trade_data_2017)

154

In [29]:
from collections import Counter


In [19]:
C_t=Counter(trade_data["Taxon"])

In [41]:
C_g=Counter(trade_data["Genus"])

In [20]:
for key, value in sorted(C_t.items(), key=lambda item: (item[1], item[0])):
    print ("%s: %s" % (key, value))

Abronia graminea: 1
Acampe praemorsa: 1
Acanthastrea rotundoflora: 1
Acanthephippium pictum: 1
Acanthocereus hirschtianus: 1
Acanthopathes spp.: 1
Acanthophippium mantinianum: 1
Accipiter brachyurus: 1
Accipiter brevipes: 1
Accipiter hybrid: 1
Accipiter minullus: 1
Accipiter soloensis: 1
Accipiter striatus: 1
Accipiter tachiro: 1
Aceros corrugatus: 1
Acianthera adamantinensis: 1
Acianthera angustisepala: 1
Acianthera butcheri: 1
Acianthera caldensis: 1
Acianthera calypso: 1
Acianthera capillaris: 1
Acianthera fockei: 1
Acianthera leptotifolia: 1
Acianthera limae: 1
Acianthera platystachys: 1
Acianthera recurva: 1
Acianthera saundersiana: 1
Acianthera scalpricaulis: 1
Acineta beyrodtiana: 1
Acineta chrysantha: 1
Acineta densa: 1
Acineta spp.: 1
Acipenser brevirostrum: 1
Acipenser fulvescens: 1
Acipenser mikadoi: 1
Acipenser nudiventris: 1
Acipenser sinensis: 1
Acipenser sturio: 1
Acriopsis indica: 1
Acropora bifurcata: 1
Acropora cardenae: 1
Acropora convexa: 1
Acropora cuneata: 1
Acrop

Eriosyce heinrichiana: 1
Eriosyce islayensis: 1
Eriosyce napina: 1
Eriosyce senilis: 1
Errina cheilopora: 1
Errina cochleata: 1
Errina gracilis: 1
Errina kerguelensis: 1
Errina reticulata: 1
Errinopora nanneca: 1
Errinopsis fenestrata: 1
Errinopsis reticulum: 1
Erycina glossomystax: 1
Erycina pumilio: 1
Erythrodes spp.: 1
Eryx tataricus: 1
Escobaria chihuahuensis: 1
Escobaria cubensis: 1
Escobaria dasyacantha: 1
Escobaria laredoi: 1
Escobaria tuberculosa: 1
Escobaria vivipara: 1
Eubalaena glacialis: 1
Eulemur macaco: 1
Eulemur mongoz: 1
Eulemur rufus: 1
Eulophia flava: 1
Eulophia herbacea: 1
Eulophia spectabilis: 1
Eulophia zollingeri: 1
Eunectes spp.: 1
Euphorbia ammak: 1
Euphorbia ampliphylla: 1
Euphorbia antiquorum: 1
Euphorbia antso: 1
Euphorbia atrox: 1
Euphorbia bayeri: 1
Euphorbia berotica: 1
Euphorbia bongolavensis: 1
Euphorbia brachyphylla: 1
Euphorbia brunellii: 1
Euphorbia bubalina: 1
Euphorbia clandestina: 1
Euphorbia clava: 1
Euphorbia cuneata: 1
Euphorbia curvirama: 1
Eup

Poicephalus fuscicollis: 1
Poicephalus spp.: 1
Polyplectron germaini: 1
Polyplectron malacense: 1
Polyplectron napoleonis: 1
Polystachya hybrid: 1
Ponthieva maculata: 1
Pontoporia blainvillei: 1
Porites annae: 1
Porites branneri: 1
Porphyrostachys spp.: 1
Porroglossum apoloae: 1
Porroglossum dreisei: 1
Porroglossum echidna: 1
Porroglossum ecuagenerense: 1
Porroglossum hystrix: 1
Porroglossum meridionale: 1
Porroglossum merinoi: 1
Porroglossum mordax: 1
Porroglossum parsonsii: 1
Porroglossum schramii: 1
Porroglossum teretilabia: 1
Prescottia spp.: 1
Prionailurus bengalensis bengalensis: 1
Prioniturus discurus: 1
Pristis pectinata: 1
Prosopeia personata: 1
Prosthechea boothiana: 1
Prosthechea calamaria: 1
Prosthechea citrina: 1
Prosthechea mariae: 1
Prosthechea tigrina: 1
Psammocora superficialis: 1
Psephotus chrysopterygius: 1
Pseudocirrhipathes spp.: 1
Pseudolaelia dutrae: 1
Pseudorhipsalis acuminata: 1
Psittacula himalayana: 1
Psittaculirostris salvadorii: 1
Pteridophora alberti: 1
Pt

Ornithocephalus dolabratus: 2
Ornithoptera akakeae: 2
Ornithoptera spp.: 2
Otolemur crassicaudatus: 2
Otostylis spp.: 2
Otus asio: 2
Otus rutilus: 2
Ovis ammon severtzovi: 2
Ovis aries cycloceros: 2
Pachycereus hollianus: 2
Pachypodium enigmaticum: 2
Pachypodium geayi: 2
Palleon nasus: 2
Pandinus dictator: 2
Pandinus imperator: 2
Panisea uniflora: 2
Panthera leo persica: 2
Pantholops hodgsonii: 2
Paphiopedilum ciliolare: 2
Paphiopedilum javanicum: 2
Paphiopedilum ooii: 2
Paphiopedilum papuanum: 2
Paphiopedilum victoria-mariae: 2
Papilionanthe uniflora: 2
Papilionanthe vandarum: 2
Pardofelis marmorata: 2
Parodia alacriportana: 2
Parodia comarapana: 2
Parodia concinna: 2
Parodia microsperma: 2
Parodia ottonis: 2
Pavona clavus: 2
Pectinia paeonia: 2
Pelexia callifera: 2
Pelophilus spp.: 2
Phaethornis bourcieri: 2
Phaethornis longirostris: 2
Phaius antoninae: 2
Phaius fragilis: 2
Phaius hybrid: 2
Phaius tancarvilliae: 2
Phalaenopsis cochlearis: 2
Phalaenopsis fimbriata: 2
Phalaenopsis mich

Rodriguezia spp.: 3
Rodriguezia venusta: 3
Rudolfiella floribunda: 3
Rupicola rupicola: 3
Sandalolitha robusta: 3
Sarkidiornis melanotos: 3
Sarracenia catesbaei: 3
Scaphosepalum andreettae: 3
Scaphosepalum breve: 3
Scaphosepalum fimbriatum: 3
Scaphosepalum gibberosum: 3
Schlumbergera orssichiana: 3
Schomburgkia splendida: 3
Schomburgkia spp.: 3
Schomburgkia superbiens: 3
Schomburgkia undulata: 3
Schomburgkia weberbaueriana: 3
Serpentes spp.: 3
Siderastrea siderea: 3
Sievekingia marsupialis: 3
Smitinandia micrantha: 3
Sobralia atropubescens: 3
Sobralia wilsoniana: 3
Solenidium spp.: 3
Sousa chinensis: 3
Spathoglottis spp.: 3
Sphaeropteris cooperi: 3
Sphaeropteris glauca: 3
Spiranthes spp.: 3
Spizaetus nipalensis: 3
Stanhopea gibbosa: 3
Stanhopea stevensonii: 3
Stanhopea tricornis: 3
Staurochilus guibertii: 3
Staurochilus ionosmus: 3
Stelis maxima: 3
Stenocactus ochoterenanus: 3
Stenocereus thurberi: 3
Stilifolium wittii: 3
Strigops habroptila: 3
Strix rufipes: 3
Strix virgata: 3
Sudamer

Macrochelys temminckii: 8
Mammillaria baumii: 8
Mammillaria carmenae: 8
Mammillaria duwei: 8
Mammillaria pectinifera: 8
Mantella madagascariensis: 8
Masdevallia davisii: 8
Masdevallia decumana: 8
Masdevallia veitchiana: 8
Maxillaria striata: 8
Meleagris ocellata: 8
Melocactus azureus: 8
Miltoniopsis hybrid: 8
Nestor notabilis: 8
Northiella haematogaster: 8
Operculicarya decaryi: 8
Orchis hybrid: 8
Paphiopedilum hangianum: 8
Paphiopedilum philippinense: 8
Paphiopedilum primulinum: 8
Phaius flavus: 8
Phaius spp.: 8
Phaius tankervilleae: 8
Phalaenopsis cornucervi: 8
Phalaenopsis lindenii: 8
Phalaenopsis parishii: 8
Phalaenopsis sumatrana: 8
Phragmipedium longifolium: 8
Pionus senilis: 8
Platycercus caledonicus: 8
Polaskia spp.: 8
Polyodon spathula: 8
Polystachya spp.: 8
Pongo abelii: 8
Potos flavus: 8
Promenaea hybrid: 8
Prosthechea spp.: 8
Python anchietae: 8
Ravenea rivularis: 8
Rebutia minuscula: 8
Rhipsalis pulchra: 8
Saguinus imperator: 8
Stelis spp.: 8
Stenocactus multicostatus: 8
S

Pilosocereus pachycladus: 42
Poicephalus senegalus: 42
Ferocactus pilosus: 43
Pachycereus pecten-aboriginum: 43
Parodia spp.: 43
Testudo graeca: 43
Acanthophyllia deshayesiana: 44
Cleistocactus strausii: 44
Ferocactus glaucescens: 44
Python spp.: 44
Distichopora spp.: 45
Eunectes notaeus: 45
Opuntia spp.: 45
Ovis ammon: 45
Acipenser stellatus: 46
Ammotragus lervia: 46
Diploastrea heliopora: 46
Espostoa lanata: 46
Felis silvestris: 46
Gastrodia elata: 46
Seriatopora hystrix: 46
Tayassu pecari: 46
Corallium japonicum: 47
Miltonia hybrid: 47
Accipiter gentilis: 48
Ceratotherium simum simum: 48
Crocodylus acutus: 48
Mammillaria zeilmanniana: 48
Schlumbergera hybrid: 48
Cereus spp.: 49
Galaxea astreata: 49
Paphiopedilum spp.: 49
Scolymia spp.: 49
Acipenser baerii x naccarii: 50
Aloe mitriformis: 50
Aloe variegata: 50
Cattleya spp.: 50
Pan troglodytes: 50
Parodia warasii: 50
Phalaenopsis amabilis: 50
Plerogyra simplex: 50
Dendrobium spp.: 51
Heliopora coerulea: 51
Pilosocereus spp.: 51
Sphae

Top 10 Taxon from 2016-2017:
Varanus niloticus: 735
Macaca fascicularis: 793
Loxodonta africana: 854
Acipenser baerii: 911
Python bivittatus: 925
Varanus salvator: 960
Crocodylus porosus: 1404
Alligator mississippiensis: 2099
Python reticulatus: 2309
Crocodylus niloticus: 2417

In [42]:
for key, value in sorted(C_g.items(), key=lambda item: (item[1], item[0])):
    print ("%s: %s" % (key, value))

Abronia: 1
Acanthopathes: 1
Acanthophippium: 1
Aceros: 1
Adenoncos: 1
Agrias: 1
Agrostophyllum: 1
Alamania: 1
Allobates: 1
Alluaudiopsis: 1
Amitostigma: 1
Andinia: 1
Anthracothorax: 1
Aphanipathes: 1
Ardeotis: 1
Argusianus: 1
Armatocereus: 1
Arpophyllum: 1
Ascochilus: 1
Ascoglossum: 1
Aspidogyne: 1
Astroides: 1
Atretium: 1
Atrophaneura: 1
Australogyra: 1
Austrocactus: 1
Baillonius: 1
Baskervilla: 1
Bathypathes: 1
Bos: 1
Brasilicereus: 1
Brasiliopuntia: 1
Bromheadia: 1
Bubulcus: 1
Budorcas: 1
Burhinus: 1
Busarellus: 1
Cacajao: 1
Calibanus: 1
Calliphlox: 1
Calloselasma: 1
Calyptocephalella: 1
Calyptopora: 1
Catreus: 1
Cattleyopsis: 1
Cephalopterus: 1
Cephalorhynchus: 1
Chrysemys: 1
Chrysolampis: 1
Chytroglossa: 1
Ciccaba: 1
Circaetus: 1
Cladocora: 1
Cladopathes: 1
Cladopsammia: 1
Coeliopsis: 1
Coleocephalocereus: 1
Colinus: 1
Colpophyllia: 1
Conopora: 1
Conradilla: 1
Constantia: 1
Corallorhiza: 1
Cordiglottis: 1
Corryocactus: 1
Corymborkis: 1
Coscinaraea: 1
Cranichis: 1
Crocodilurus: 1
C

Top 10 Genus from 2016-2017:
Aloe: 935
Dendrobium: 970
Macaca: 971
Euphorbia: 1147
nan: 1459
Caiman: 1627
Acipenser: 1976
Varanus: 2079
Alligator: 2100
Python: 4248
Crocodylus: 4408

In [30]:
Cf=Counter(trade_data["Family"])

In [31]:
for key, value in sorted(Cf.items(), key=lambda item: (item[1], item[0])):
    print ("%s: %s" % (key, value))

Anguidae: 1
Aromobatidae: 1
Astrocoeniidae: 1
Bufonidae: 1
Burhinidae: 1
Calyptocephalellidae: 1
Cladopathidae: 1
Dioscoreaceae: 1
Fungiacyathidae: 1
Gavialidae: 1
Leiopathidae: 1
Lycopodiaceae: 1
Menyanthaceae: 1
Sagittariidae: 1
Sciaenidae: 1
Sciuridae: 1
Stangeriaceae: 1
Thylacomyidae: 1
Trogonidae: 1
Tropidophiidae: 1
Unionidae: 1
Ambystomatidae: 2
Antilocapridae: 2
Araceae: 2
Ardeidae: 2
Balaenicipitidae: 2
Cryptobranchidae: 2
Dasyproctidae: 2
Dicroglossidae: 2
Dugongidae: 2
Gerrhosauridae: 2
Indriidae: 2
Iniidae: 2
Lacertidae: 2
Latimeriidae: 2
Parmeliaceae: 2
Rhincodontidae: 2
Rhizangiidae: 2
Salamandridae: 2
Scolopacidae: 2
Selaginellaceae: 2
Tachyglossidae: 2
Aphanipathidae: 3
Bradypodidae: 3
Hydrophiidae: 3
Hystricidae: 3
Phalangeridae: 3
Pycnonotidae: 3
Schizopathidae: 3
Tarsiidae: 3
Carettochelyidae: 4
Cotingidae: 4
Dermochelyidae: 4
Emberizidae: 4
Flabellidae: 4
Fringillidae: 4
Meandriniidae: 4
Neoceratodontidae: 4
Nymphalidae: 4
Scorpionidae: 4
Scrophulariaceae: 4
Ericace

Top 10 Family from 2016-2017:
Felidae: 1408
Cercopithecidae: 1646
Varanidae: 2079
Acipenseridae: 2341
Psittacidae: 3611
Alligatoridae: 3758
Crocodylidae: 4423
Pythonidae: 4488
Cactaceae: 4800
Orchidaceae: 9973

In [49]:
C_term=Counter(trade_data["Term"])

In [52]:
for key, value in sorted(C_term.items(), key=lambda item: (item[1], item[0])):
    print ("%s: %s" % (key, value))

fruit: 1
gall: 1
horn pieces: 2
swim bladders: 2
gall bladders: 3
coral sand: 4
eggshell: 4
fingerlings: 5
trunk: 5
genitalia: 9
pupae: 9
baleen: 10
horn carvings: 11
scales: 12
fur products (large): 15
bone pieces: 17
musk: 17
bark: 18
carapaces: 18
fibres: 18
skeletons: 19
bone carvings: 21
ears: 22
flowers: 24
plates: 24
horns: 29
jewellery - ivory : 29
pearl: 30
fur product (small): 34
veneer: 34
ivory pieces: 39
unspecified: 41
hair products: 44
claws: 53
tails: 53
sides: 54
rug: 59
fins: 63
eggs (live): 67
feet: 69
piano keys: 71
timber: 87
stems: 89
bones: 90
cloth: 90
logs: 106
powder: 113
hair: 120
dried plants: 127
wood product: 139
shells: 144
wax: 147
teeth: 175
leaves: 184
tusks: 188
seeds: 192
eggs: 220
chips: 225
sawn wood: 238
oil: 243
cultures: 266
cosmetics: 297
meat: 330
feathers: 336
roots: 353
medicine: 380
jewellery: 386
skulls: 396
derivatives: 454
ivory carvings: 540
carvings: 693
skin pieces: 748
leather products (large): 866
bodies: 915
extract: 965
caviar: 12

Top 10 terms traded from 2016 to 2017:
bodies: 915
extract: 965
caviar: 1206
garments: 1456
trophies: 1603
raw corals: 1903
skins: 2271
specimens: 2852
leather products (small): 10106
live: 33862

In [37]:
country_data=pd.read_excel("ISO 3166-1 alpha-2.xlsx")

In [38]:
len(country_data["alpha-2"])

249

In [42]:

for i in range(len(trade_data["Importer"])):
    for e in range(len(country_data["alpha-2"])):
        if trade_data["Importer"][i]==country_data["alpha-2"][e]:
            #im_country.append(country_data["name"].values[e])
            trade_data["Importer"]=trade_data["Importer"].replace(trade_data["Importer"][i],country_data["name"].values[e])

In [43]:
trade_data

Year App.                     Taxon        Class             Order  \
0      2016    I            Aquila heliaca         Aves     Falconiformes   
1      2016    I            Aquila heliaca         Aves     Falconiformes   
2      2016    I      Haliaeetus albicilla         Aves     Falconiformes   
3      2016    I      Haliaeetus albicilla         Aves     Falconiformes   
4      2016    I      Haliaeetus albicilla         Aves     Falconiformes   
5      2016    I      Haliaeetus albicilla         Aves     Falconiformes   
6      2016    I            Harpia harpyja         Aves     Falconiformes   
7      2016    I            Harpia harpyja         Aves     Falconiformes   
8      2016    I            Harpia harpyja         Aves     Falconiformes   
9      2016    I    Acipenser brevirostrum  Actinopteri  Acipenseriformes   
10     2016    I          Acipenser sturio  Actinopteri  Acipenseriformes   
11     2016    I           Ailurus fulgens     Mammalia         Carnivora   
12     2016    I           Ailurus fulgens     Mammalia         Carnivora   
13     2016    I           Ailurus fulgens     Mammalia         Carnivora   
14     2016    I           Ailurus fulgens     Mammalia         Carnivora   
15     2016    I           Ailurus fulgens     Mammalia         Carnivora   
16     2016    I           Ailurus fulgens     Mammalia         Carnivora   
17     2016    I           Ailurus fulgens     Mammalia         Carnivora   
18     2016    I           Ailurus fulgens     Mammalia         Carnivora   
19     2016    I           Ailurus fulgens     Mammalia         Carnivora   
20     2016    I        Alligator sinensis     Reptilia        Crocodylia   
21     2016    I        Melanosuchus niger     Reptilia        Crocodylia   
22     2016    I          Anas laysanensis         Aves      Anseriformes   
23     2016    I          Anas laysanensis         Aves      Anseriformes   
24     2016    I      Asarcornis scutulata         Aves      Anseriformes   
25     2016    I      Asarcornis scutulata         Aves      Anseriformes   
26     2016    I       Branta sandvicensis         Aves      Anseriformes   
27     2016    I       Branta sandvicensis         Aves      Anseriformes   
28     2016    I       Branta sandvicensis         Aves      Anseriformes   
29     2016    I       Branta sandvicensis         Aves      Anseriformes   
...     ...  ...                       ...          ...               ...   
67131  2017   II     Stigmochelys pardalis     Reptilia        Testudines   
67132  2017   II         Aquilaria filaria          NaN          Myrtales   
67133  2017   II        Amazilia fimbriata         Aves       Apodiformes   
67134  2017   II     Chlorostilbon notatus         Aves       Apodiformes   
67135  2017   II       Florisuga mellivora         Aves       Apodiformes   
67136  2017   II          Glaucis hirsutus         Aves       Apodiformes   
67137  2017   II       Heliodoxa aurescens         Aves       Apodiformes   
67138  2017   II         Hylocharis cyanus         Aves       Apodiformes   
67139  2017   II    Leucippus chlorocercus         Aves       Apodiformes   
67140  2017   II  Phaethornis atrimentalis         Aves       Apodiformes   
67141  2017   II     Phaethornis bourcieri         Aves       Apodiformes   
67142  2017   II      Phaethornis hispidus         Aves       Apodiformes   
67143  2017   II       Phaethornis malaris         Aves       Apodiformes   
67144  2017   II         Phaethornis ruber         Aves       Apodiformes   
67145  2017   II        Thalurania furcata         Aves       Apodiformes   
67146  2017   II           Threnetes niger         Aves       Apodiformes   
67147  2017   II              Ursus arctos     Mammalia         Carnivora   
67148  2017   II         Varanus niloticus     Reptilia            Sauria   
67149  2017   II          Varanus salvator     Reptilia            Sauria   
67150  2017   II          Varanus salvator     Reptilia            Sauria   
671

In [44]:
for i in range(len(trade_data["Exporter"])):
    for e in range(len(country_data["alpha-2"])):
        if trade_data["Exporter"][i]==country_data["alpha-2"][e]:
            #im_country.append(country_data["name"].values[e])
            trade_data["Exporter"]=trade_data["Exporter"].replace(trade_data["Exporter"][i],country_data["name"].values[e])

In [45]:
trade_data["Exporter"]

0                                              Netherlands
1                                                   Serbia
2                                                   Norway
3                                                   Norway
4                                                  Iceland
5                                                   Serbia
6                                                   France
7                                                   France
8                                                   France
9                                                  Germany
10                              Iran (Islamic Republic of)
11                                             New Zealand
12                                United States of America
13                                                 Germany
14                                United States of America
15                                United States of America
16                                                   Chi

In [46]:
len(trade_data["Importer"])

67161

In [43]:
C_Im=Counter(trade_data["Importer"])

In [44]:
for key, value in sorted(C_Im.items(), key=lambda item: (item[1], item[0])):
    print ("%s: %s" % (key, value))

Anguilla: 1
Burkina Faso: 1
Central African Republic: 1
Dominica: 1
Gambia: 1
Kiribati: 1
Samoa: 1
Turks and Caicos Islands: 1
Chad: 2
Djibouti: 2
Equatorial Guinea: 2
French Southern Territories: 2
Guinea: 2
Guyana: 2
Liberia: 2
Mayotte: 2
Saint Vincent and the Grenadines: 2
Timor-Leste: 2
Tokelau: 2
Virgin Islands (U.S.): 2
Belize: 3
Eritrea: 3
Mali: 3
Papua New Guinea: 3
Rwanda: 3
Yemen: 3
Benin: 4
Bolivia (Plurinational State of): 4
Botswana: 4
Cocos (Keeling) Islands: 4
Grenada: 4
Niger: 4
Syrian Arab Republic: 4
Haiti: 5
Mauritania: 5
Togo: 5
YU: 5
Holy See: 6
Isle of Man: 6
Malta: 6
Palau: 6
Seychelles: 6
Cabo Verde: 7
SU: 7
Saint Lucia: 7
Northern Mariana Islands: 8
Swaziland: 8
Zambia: 8
Cuba: 9
Jamaica: 9
Malawi: 9
Fiji: 10
Gabon: 10
Greenland: 10
Liechtenstein: 10
Puerto Rico: 10
Ethiopia: 11
Honduras: 11
Cameroon: 12
Croatia: 12
Mozambique: 12
Nicaragua: 13
Saint Kitts and Nevis: 14
Sudan: 14
Tanzania, United Republic of: 14
Antigua and Barbuda: 15
Virgin Islands (British):

Top 10 importer from 2016-2017:
Canada: 1409
United Arab Emirates: 1506
Singapore: 1576
China: 2126
Switzerland: 2508
Hong Kong: 2555
France: 2604
Germany: 4913
Japan: 5192
United States of America: 9722

In [ ]:
C_ex=Counter(trade_data["Exporter"])

In [56]:
for key, value in sorted(C_ex.items(), key=lambda item: (item[1], item[0])):
    print ("%s: %s" % (key, value))

AN: 1
Bosnia and Herzegovina: 1
DD: 1
Faroe Islands: 1
Guadeloupe: 1
Isle of Man: 1
Lesotho: 1
Niger: 1
Samoa: 1
Swaziland: 1
Timor-Leste: 1
Tokelau: 1
Burundi: 2
Comoros: 2
Cyprus: 2
Eritrea: 2
Gibraltar: 2
Liechtenstein: 2
Andorra: 3
Aruba: 3
Botswana: 3
Cabo Verde: 3
Chad: 3
Kiribati: 3
Korea (Democratic People's Republic of): 3
Montenegro: 3
Sierra Leone: 3
XV: 3
Bangladesh: 4
Guinea: 4
Iraq: 4
Saint Barthélemy: 4
Saint Vincent and the Grenadines: 4
South Georgia and the South Sandwich Islands: 4
Vanuatu: 4
Saint Helena, Ascension and Tristan da Cunha: 5
Yemen: 5
Grenada: 6
Saint Lucia: 6
Sao Tome and Principe: 6
Dominica: 7
Gambia: 7
Guinea-Bissau: 7
Maldives: 7
Albania: 8
Antarctica: 8
Egypt: 8
French Polynesia: 8
Sint Maarten (Dutch part): 8
Falkland Islands (Malvinas): 9
Azerbaijan: 10
British Indian Ocean Territory: 10
Moldova (Republic of): 10
Syrian Arab Republic: 10
Angola: 11
Antigua and Barbuda: 11
Central African Republic: 11
Equatorial Guinea: 11
Gabon: 11
Liberia: 11
T

Top 10 Exporter from 2016-2017:
CH: 1490
SG: 1497
EC: 1501
CN: 1506
DE: 2653
FR: 3745
US: 4503
IT: 5126
ID: 6526
NL: 7201

B Breeding in captivity or artificial propagation
E Educational
G Botanical garden
H Hunting trophy
L Law enforcement / judicial / forensic
M Medical (including biomedical research)
N Reintroduction or introduction into the wild
P Personal
Q Circus or travelling exhibition
S Scientific
T Commercial
Z Zoo

In [40]:
C_purpose=Counter(trade_data["Purpose"])

In [47]:
C_purpose

Counter({'T': 47081,
         'Q': 1054,
         'S': 2382,
         'P': 5859,
         'Z': 776,
         'B': 384,
         'E': 323,
         'H': 2145,
         'N': 50,
         'L': 90,
         'M': 867,
         nan: 6059,
         'G': 91})